.github/workflows/testground.yml

```diff
@@ -0,0 +1,59 @@
+name: Run Testground
+
+on:
+  push:
+jobs:
+  run-testground:
+      runs-on: ubuntu-latest
+      steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "^1.19.0"
+      # Get testground and our test plan code
+      - uses: actions/checkout@v3
+        with:
+         repository: 'statechannels/testground'
+         path: "code/testground"
+      - uses: actions/checkout@v3
+        with:
+          repository: 'statechannels/go-nitro-testground'
+          path: "code/go-nitro-testground"
+          ref: update-to-main #TODO Remove when https://github.com/statechannels/go-nitro-testground/pull/103 is merged
+      
+      # Set up caching for docker (which testground uses) and go modules
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+
+      # Build the testground daemon
+      - name: Build
+        run: make install
+        working-directory: "code/testground"
+      # Update our test plan so it uses the version of go-nitro from this workflow
+      - name: Update Test Dependency
+        run: go get github.com/statechannels/go-nitro@${{github.sha}}
+        working-directory: "code/go-nitro-testground"
+
+      - name: Import Test 
+        run:    testground plan import --from ./go-nitro-testground
+        working-directory: "code"
+      # Run a short test using the wait flag so we block until it completes
+      - name: Run Test 
+        run: | 
+         testground --endpoint=http://34.168.92.245:8042  run s --wait \
```

Comment:
Will the github action fail if the testground task fails? I am guessing it will?

Trying it out locally, the `--wait` flag causes some logs to be streamed back to the console, which should be useful. And if the task fails to launch, we definitely get a nonzero exit code.

Reply:
Yup, if the testground task fails the test will fail as well.

> This is great!
> 
> It occurs to me that if a PR changes the go-nitro API, the testground job will likely fail, and we will need to go through the rigamarole of updating the other repo's test script. Is there an argument for pulling that test script (at least) into this repo to make it easier to keep things in sync and prevent breaking changes?

Yup, changing the API is slightly painful as you'll have to update  `go-nitro-testground` before the job passes. I think it would be nice to pull in `go-nitro-testground` into `go-nitro`. The only catch is that `go-nitro-testground` is it's own module, which would mean we'd hav two go modules in one repo if we move it to `go-nitro`. Maybe something like [workspaces](https://go.dev/ref/mod#workspaces) could help there?